LaTeX/TeX.sublime-syntax

```diff
@@ -175,34 +175,54 @@ contexts:
     - include: paragraph-pop
 
   tex-constant-value:
-    - match: ((-?)\d+(\.)\d*)({{dimunits}}?)
-      scope: meta.number.float.decimal.tex
-      captures:
-        1: constant.numeric.value.tex
-        2: keyword.operator.arithmetic.tex
-        3: punctuation.separator.decimal.tex
-        4: constant.numeric.suffix.tex
-    - match: ((-?)\d+)({{dimunits}}?)
-      scope: meta.number.integer.decimal.tex
-      captures:
-        1: constant.numeric.value.tex
-        2: keyword.operator.arithmetic.tex
-        3: constant.numeric.suffix.tex
+    - match: (-?)\d+(\.)\d*
+      scope: constant.numeric.value.tex
+      captures:
+        1: keyword.operator.arithmetic.tex
+        2: punctuation.separator.decimal.tex
+      push:
+        - decimal-float-meta
+        - tex-dimension-suffix
```

Comment:
Can units appear on the next line?

```
\foo=12
   pt
```

If not I'd probably keep going with capture groups

Reply:
from the point of TeX, a single newline is (mostly) the same as a regular space, so yes.